x/feemarket/keeper/eip1559.go

```diff
@@ -25,13 +25,13 @@ func (k Keeper) CalculateBaseFee(ctx sdk.Context) *big.Int {
 
 	// If the current block is the first EIP-1559 block, return the InitialBaseFee.
 	if ctx.BlockHeight() == params.EnableHeight {
-		return new(big.Int).SetInt64(params.InitialBaseFee)
+		return k.GetBaseFee(ctx)
 	}
 
 	// get the block gas used and the base fee values for the parent block.
 	parentBaseFee := k.GetBaseFee(ctx)
 	if parentBaseFee == nil {
-		parentBaseFee = new(big.Int).SetInt64(params.InitialBaseFee)
+		return nil
```

Comment:
why this change?

Reply:
does it make sense to return `params.InitialBaseFee` here?  

if parentbasefee cannot be found, should the function calculate return nil and disable basefee?

Reply:
the default behavior on geth is to return the initial base fee afaik, if the base fee is not found then it we can assume it should have been checked before?

Reply:
go-ethereum checks this in the `VerifyEip1559Header ` function
https://github.com/ethereum/go-ethereum/blob/53b1420edefcf6be154ea6df887b1e4d68fcc36f/consensus/misc/eip1559.go#L41-L44, as part of the `verifyHeader` function - e.g. in https://github.com/ethereum/go-ethereum/blob/b3b8b268eb585dfd3c1c9e9bbebc55968f3bec4b/consensus/ethash/consensus.go#L300-L303

Are block headers checked by ethermint or geth?

Reply:
afaik, we don't validate `header.basefee` on london block on ethermint.

Geth uses a strong assumption that parent base fee cannot be nil thanks to the validation, thus it is not checked during the calculation
https://github.com/ethereum/go-ethereum/blob/53b1420edefcf6be154ea6df887b1e4d68fcc36f/consensus/misc/eip1559.go#L68

 But in our case, we have different config for enabling feemarket and setting london blocks number which are not necessarily corollated by the code itself.
 
 I guess we need to make a decision in that case. 
 
 I can return the "param.baseFee" here if it is the behavior that we want to have, let me know what you think

Reply:
we can return the default param base fee value for consistency with geth

Reply:
actually since we store the base fee as a module param, this line does not make more sense anymore

since `k.GetBaseFee(ctx)` and `paramsBaseFee` return the same things.

do you means we should 
- either use 0 value for parentBaseFee
- introduce a new parameter in genesis default_base_fee and use it here?